### ISSUE TITLE
Using `shared` within a `views` function - behaviour change between v6 and v8

### ISSUE THREAD

**Issue by andybee** [architect/architect#1059]
Apologies if this was discussed at the time of release in either v7 or v8, but I can't find reference to anything obvious that would cause this change in behaviour.

Inside the shared `views` folder I have a template function in turn depending on a function within `shared`. Previously this used to work, but since moving to v8 I get a "Cannot find module" error.

My knowledge of commonjs' module resolution is a little rusty, but I'd have thought when `views` is used within an actual `@http` handler function it should be able to import `@architect/shared` without issue? Is this a Sandbox/paths issue...?

**Comment by andybee** [comment]
I've created a quick demo of the behavioural difference v6 vs v8:
https://github.com/andybee/arc-v6-v8-import-issue

If you go in the v6 folder and run `npm i && npx arc sandbox`, then hit `localhost:3333` you'll see a very dull basic HTML response. If you do the same in the v8 folder you'll see an import error. The codebase is identical for each and simulates the arrangement in my own project I'm having trouble with - function depends on `@architect/views`, the code within which depends on `@architect/shared`.

**Comment by andybee** [comment]
Tracked this down, the change is the use of symlinks over copying the folders each time a change is detected. When using symlinks, Node's default behaviour is to resolve the symlink and use the "true" path instead (so you can do kooky stuff like have multiple symlinks to the same module only keep one module in the cache, etc.) But once resolved, it impacts the way that module loads it's own dependencies.

TL;DR - this node behaviour, in conjunction with symlinks, means the behaviour of Sandbox differs from that of a deployed Lambda function where the shared and views folders are copied across.

I have found a way Sandbox could potentially fix this, but I'm a little concerned by the caveats Node.js warn you about with it. I'll pop a proper PR on here shortly.

**Comment by andybee** [comment]
Here's the PR: https://github.com/architect/sandbox/pull/542

**Comment by ryanblock** [comment]
@andybee super sorry it's taken me so long to get caught up on this (and other) Sandbox PRs!

So the problem with the solution in the PR is that the `--preserve-symlinks` flag appears to be interfering with the module resolution checks we do to ensure all dependencies are installed.

I started to look into it but don't think I'll have time to fully tease it apart today. So I don't yet know if that flag will work yet.

That said, there is a workaround: modules in `views` or `shared` can call one another in production relatively, e.g. `src/views/foo.js` can call `src/shared/bar.js` with `require('../shared/bar')` because once copied into place for production, `shared` and `views` both live within `node_modules/@architect/`.

How do you want to proceed from here?

**Comment by andybee** [comment]
@ryanblock No worries.

Since end of January I've been doing just that, always using relative paths.

I can't work out if this is a potential gotcha for new users or not (and therefore worth of trying to "fix"), that everywhere *apart from* this one place you do A but here you need workaround B?
